### PR TITLE
"data info pub/pkg" command

### PR DESCRIPTION
**Pull request by anuveyatsu**
* setups - marked-terminal, renaming test file
* `/lib/info.js` logic for getting metadata and readme, then logging to console
  * tests for it including fixture files
* `/bin/data-info.js` help messages and running command

P.S. This does not include spinner and progress bar - we'll implement it in #48

### CODE REVIEW

File: lib/utils/marked.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+var marked = require('marked')
+var TerminalRenderer = require('marked-terminal')
```
Comment (zelima) [inline lib/utils/marked.js]: let / const?

File: lib/info.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  // console.log full readme
+  console.log(customMarked(`# README\n${readme}\n***`))
+}
+
+const getReadme = async (url) => {
+  let res = await axios.get(url).catch(err => {
+    logger(err.message, 'error', true)
+  })
+  if(!res) {
```
Comment (zelima) [inline lib/info.js]: Don't think the response will be undefined if we are handling the error and exiting up there. so think this check is extra here

**Review by Mikanebu — APPROVED** [review]

File: lib/info.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+const axios = require('axios')
+const urljoin = require('url-join')
+const {getMetadata} = require('./get.js')
```
Comment (zelima) [inline lib/info.js]: Hm... not for this, but maybe it's worth taking getMetadata into utils?

File: lib/info.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+const {logger} = require('./utils/log-handler.js')
+const utils = require('../lib/utils/common')
+
+const getInfo = async (dhpkgid) => {
+  const { publisher, pkg } = utils.parseIdentifier(dhpkgid)
+
+  const metadata = await getMetadata(publisher, pkg)
+  const readme = await getReadme(urljoin(metadata.bitstore_url, 'README.md'))
+
+  const firstParagraphReadme = `# ABOUT\n${readme.substring(0, 200)}... *see more below*`
```
Comment (zelima) [inline lib/info.js]: first 200? what if in the middle of the word? maybe 
`readme.substring(0, 200).split(' ').pop().join(' ') ...` would be better

File: lib/info.js [hunk 5]
```diff
@@ -13,10 +13,14 @@ const getInfo = async (dhpkgid) => {
...
-  let resourcesInfo = '\n# RESOURCES\n'
+  let resourcesInfo = `
+  \n# RESOURCES\n
+  | Name | Format |
+  |------|--------|
+  `
 
   metadata.descriptor.resources.forEach(resource => {
-    resourcesInfo = `${resourcesInfo}* **${resource.name}** [${resource.format}]\n`
+    resourcesInfo += `|${resource.name}|${resource.format}|\n`
```
Comment (zelima) [inline lib/info.js]: can not this be done with markdown table?  would look better

File: bin/data-info.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+  ${underline('Usage:')}
+
+    ${chalk.gray('#')} Get Data Package info for given publisher and package:
+    ${boldText(`$ data info <publisher>/<package>`)}
+
+  ${underline('Example:')}
+
+    ${chalk.gray('#')} Get Data Package info for given core publisher and co2-ppm package:
+    ${boldText(`$ data info core/co2-ppm`)}
```
Comment (zelima) [inline bin/data-info.js]: can we use `core/finance-vix` everywhere? or `gdp` maybe

File: lib/info.js (hunk 5 above)
Comment (zelima) [reply lib/info.js]: Actually, just tried and this looks great!

Comment (zelima) [reply lib/info.js]: Only thing here is that if resource does not have format it displays undefined
resources

`resource.format || ''N/A'` would fix this

**Review by zelima — COMMENTED** [review]
@anuveyatsu Awsome work, just a minor comments

File: lib/utils/marked.js (hunk 1 above)
Comment (anuveyatsu) [reply lib/utils/marked.js]: ah yes, that was ready snippet, will fix it

File: lib/info.js (hunk 3 above)
Comment (anuveyatsu) [reply lib/info.js]: @zelima yes, exactly. I just didn't want to write that function again.

File: lib/info.js (hunk 4 above)
Comment (anuveyatsu) [reply lib/info.js]: @zelima ok, let me fix that

File: lib/info.js (hunk 2 above)
Comment (anuveyatsu) [reply lib/info.js]: oh yes

File: bin/data-info.js (hunk 6 above)
Comment (anuveyatsu) [reply bin/data-info.js]: @zelima just wanted to have multiple resources so the table looks bigger for demonstration

File: lib/info.js (hunk 4 above)
Comment (zelima) [reply lib/info.js]: Hm, now I'm thinking... what if there is no README at all?
We need one more check `if (readme) { ... }`

Comment (anuveyatsu) [reply lib/info.js]: @zelima fixed that as well

File: lib/info.js [hunk 7]
```diff
@@ -9,9 +9,17 @@ const getInfo = async (dhpkgid) => {
   const { publisher, pkg } = utils.parseIdentifier(dhpkgid)
 
   const metadata = await getMetadata(publisher, pkg)
-  const readme = await getReadme(urljoin(metadata.bitstore_url, 'README.md'))
+  let readme = await getReadme(urljoin(metadata.bitstore_url, 'README.md'))
```
Comment (zelima) [inline lib/info.js]: `README.md` might now always be the case, we accept README and README.txt as well. We need to think more about this

**Pull request merged**